frontend/src/Editor/Components/Timer.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from 'react';
+
+export const Timer = function Timer({ height, properties = {}, styles, setExposedVariable, fireEvent }) {
+  const getTimeObj = (HH, MM, SS, MS) => {
+    return {
+      hour: isNaN(HH) ? 0 : parseInt(HH, 10),
+      minute: !isNaN(MM) && MM <= 59 ? parseInt(MM, 10) : 0,
+      second: !isNaN(SS) && SS <= 59 ? parseInt(SS, 10) : 0,
+      mSecond: !isNaN(MS) && MS <= 999 ? parseInt(MS, 10) : 0,
+    };
+  };
+  const [HH, MM, SS, MS] = (properties.value && properties.value.split(':')) || [];
+  const [time, setTime] = useState(getTimeObj(HH, MM, SS, MS));
+  const [state, setState] = useState('initial');
+  const [intervalId, setIntervalId] = useState(0);
+
+  useEffect(() => {
+    if (
+      properties.type === 'countDown' &&
+      time.mSecond === 0 &&
+      time.second === 0 &&
+      time.minute === 0 &&
+      time.hour === 0
+    ) {
+      intervalId && clearInterval(intervalId);
+      fireEvent('onCountDownFinish');
+    }
+  }, [time]);
+
+  useEffect(() => {
+    return () => {
+      intervalId && clearInterval(intervalId);
+    };
+  }, [intervalId]);
+
+  useEffect(() => {
+    const [HH, MM, SS, MS] = (properties.value && properties.value.split(':')) || [];
+    setTime(getTimeObj(HH, MM, SS, MS));
+  }, [properties.value]);
+
+  const onReset = () => {
+    intervalId && clearInterval(intervalId);
+    setTime(getTimeObj());
+    fireEvent('onReset');
+    setState('initial');
+    setExposedVariable('value', time);
+  };
+
+  const onStart = () => {
+    setIntervalId(
+      setInterval(() => {
```

Comment:
Why are we adding and subtracting with 15 on ms?

Reply:
`setInterval` interval is set to 15 ms

Reply:
Sorry, why that particular value of 15ms ? 🤔

Reply:
I suppose it is to make it easier to see the updation?